islandora-connector-derivative/src/main/java/ca/islandora/alpaca/connector/derivative/DerivativeOptions.java

```diff
@@ -0,0 +1,190 @@
+          final String finalInput = addJmsOptions(addBrokerName(input), concurrentConsumers, maxConcurrentConsumers);
+          // Add connectionClose and other http options.
+          final String finalOutput = addHttpOptions(output);
+          camelContext.addRoutes(new DerivativeConnector(serviceName, finalInput, finalOutput, this));
+        } catch (final Exception e) {
+          e.printStackTrace();
+        }
+      } else {
+        final StringBuilder message = new StringBuilder();
+        if (input.isBlank()) {
+          message.append(inputProperty(serviceName)).append(" is blank");
+        }
+        if (output.isBlank()) {
+          if (message.length() > 0) {
+            message.append(" and ");
+          }
+          message.append(outputProperty(serviceName)).append(" is blank");
+        }
+        message.append(", skipping");
+        LOGGER.debug(message.toString());
+      }
+
+  }
+
+  /**
+   * Just adds the JMS broker name to the provided queue/topic.
+   * @param queueName the provided queue/topic.
+   * @return the full endpoint including the broker name.
+   */
+  private String addBrokerName(final String queueName) {
+    return JMS_ENDPOINT_NAME + ":" + queueName;
+  }
+
+  /**
+   * Return the expected enabled property
+   * @param systemName the derivative system name
+   * @return the property
+   */
+  private String enabledProperty(final String systemName) {
+    return DERIVATIVE_PREFIX + "." + systemName + "." + DERIVATIVE_ENABLED_PROPERTY;
+  }
+
+  /**
+   * Return the expected input topic/queue property
+   * @param systemName the derivative system name
+   * @return the property
+   */
+  private String inputProperty(final String systemName) {
+    return DERIVATIVE_PREFIX + "." + systemName + "." + DERIVATIVE_INPUT_PROPERTY;
+  }
+
+  /**
+   * Return the expected output service url property
+   * @param systemName the derivative system name
+   * @return the property
+   */
+  private String outputProperty(final String systemName) {
+    return DERIVATIVE_PREFIX + "." + systemName + "." + DERIVATIVE_OUTPUT_PROPERTY;
+  }
+
+  /**
+   * Return the expected concurrent consumers property.
+   * @param systemName the derivative system name
+   * @return the property
+   */
+  private String concurrentConsumerProperty(final String systemName) {
+    return DERIVATIVE_PREFIX + "." + systemName + "." + DERIVATIVE_CONCURRENT_PROPERTY;
+  }
+
+  /**
+   * Return the expected max-concurrent consumers property.
+   * @param systemName the derivative system name
+   * @return the property
+   */
+  private String maxConcurrentConsumerProperty(final String systemName) {
+    return DERIVATIVE_PREFIX + "." + systemName + "." + DERIVATIVE_MAX_CONCURRENT_PROPERTY;
+  }
+
+  @Override
+  public void setApplicationContext(final ApplicationContext applicationContext) throws BeansException {
+    final String[] names = applicationContext.getBeanNamesForType(CamelContext.class);
+    if (names.length == 1) {
```

Comment:
My understanding is that in a runnable jar Camel app you only [want a single context](https://camel.apache.org/manual/faq/why-use-multiple-camelcontext.html) but I'll change this to something like
```
private static final int MIN_CAMEL_CONTEXTS = 1;
```
```
if (names.length >= MIN_CAMEL_CONTEXTS) {
```
just in case.

Reply:
Resolved in https://github.com/Islandora/Alpaca/pull/82/commits/5e5d11e3292a7c6458442aacf47b018476640f8a